doc/doc-site/docs/off-chain/dss/gcp.md

```diff
@@ -0,0 +1,916 @@
+
+   ```sh
+   AGGREGATOR_ID=$(gcloud compute instances describe aggregator \
+       --zone $ZONE \
+       --format="value(id)"
+   )
+   gcloud logging read "resource.type=gce_instance AND \
+       logName=projects/$PROJECT_ID/logs/cos_containers AND \
+       resource.labels.instance_id=$AGGREGATOR_ID" \
+       --limit 5
+   ```
+
+1. Once the aggregator has had enough time to aggregate events, check some aggregated data.
+   For example on testnet:
+
+   ```sh
+   echo $REST_URL
+   ```
+
+   ```sh
+   curl "$(printf '%s' \
+       "$REST_URL/"\
+       "limit_orders?"\
+       "order=price.desc,"\
+       "last_increase_stamp.asc&"\
+       "market_id=eq.3&"\
+       "side=eq.ask&"\
+       "order_status=eq.closed&"\
+       "limit=3"\
+   )"
+   ```
+
+### Deploy WebSockets API
+
+1. Create a connector:
+
+   ```sh
+   gcloud compute networks vpc-access connectors create \
+       websockets \
+       --range 10.64.0.0/28 \
+       --region $REGION
+   ```
+
+1. Verify connector readiness:
+
+   ```sh
+   STATE=$(gcloud compute networks vpc-access connectors describe \
+       websockets \
+       --region $REGION \
+       --format "value(state)"
+   )
+   echo "Connector state is: $STATE"
+   ```
+
+1. Construct WebSockets connection string:
+
+   ```sh
+   PGWS_DB_URI="$(printf '%s' postgres://\
+       $ADMIN_NAME:\
+       $ADMIN_PASSWORD@\
+       $POSTGRES_INTERNAL_IP/econia
+   )"
+   echo $PGWS_DB_URI
+   ```
+
+1. Deploy the `websockets` service:
+
+   ```sh
+   gcloud run deploy websockets \
+       --allow-unauthenticated \
+       --image \
+           $REGION-docker.pkg.dev/$PROJECT_ID/images/websockets:latest \
+       --port 3000 \
+       --set-env-vars "$(printf '%s' \
+           PGWS_DB_URI=$PGWS_DB_URI,\
+           PGWS_JWT_SECRET=econia_0000000000000000000000000,\
+           PGWS_CHECK_LISTENER_INTERVAL=1000,\
+           PGWS_LISTEN_CHANNEL=econiaws\
+       )" \
+       --vpc-connector websockets
+   ```
+
+1. Store service URL:
+
+   ```sh
+   WS_HTTPS_URL=$(
+       gcloud run services describe websockets \
+           --format "value(status.url)"
+   )
+   export WS_URL=$(echo $WS_HTTPS_URL | sed 's/https/wss/')
+   echo $WS_URL
+   ```
+
+1. Monitor events using the [WebSockets listening script](./websocket.md):
+
+   ```sh
+   echo $WS_URL
+   echo $REST_URL
+   echo $WS_CHANNEL
+   ```
+
+   ```sh
+   cd econia/src/python/sdk
+   poetry install
+   poetry run event
+   ```
+
+   ```sh
+   # To quit
+   <Ctrl+C>
+   ```
+
+   ```
+   cd ../../../..
+   ```
+
+## Redeployment
+
+Once you have the DSS running you might want to redeploy within the same GCP project, for example using a different chain or with new image binaries.
+
+Whenever you redeploy, restart all instances and services, and reset the database as follows so that you do not break startup dependencies or generate any corrupted data.
+
+### Basic
+
+1. [Rebuild images](#build-images) in the existing `images` registry.
```

Comment:
Is there a way we can write these instructions so that redeployment involves no downtime?
Understandable if it's too difficult for a tutorial.

Reply:
With more CI, perhaps, but yes too much for the present tutorial